noir-projects/noir-protocol-circuits/crates/types/src/abis/kernel_circuit_public_inputs/vm_circuit_public_inputs.nr

```diff
@@ -0,0 +1,126 @@
+use crate::{
+    abis::{
+    accumulated_data::{PublicAccumulatedData, PublicAccumulatedDataArrayLengths},
+    combined_constant_data::CombinedConstantData, gas::Gas, public_call_request::PublicCallRequest,
+    public_inner_call_request::PublicInnerCallRequest,
+    validation_requests::{PublicValidationRequests, PublicValidationRequestArrayLengths}
+},
+    constants::{MAX_PUBLIC_CALL_STACK_LENGTH_PER_TX, VM_CIRCUIT_PUBLIC_INPUTS_LENGTH},
+    traits::{Deserialize, Empty, Serialize}, utils::reader::Reader
+};
+
+struct VMCircuitPublicInputs {
+    constants: CombinedConstantData,
+    call_request: PublicCallRequest,
+
+    // TODO(#7124): To be deprecated.
+    public_call_stack: [PublicInnerCallRequest; MAX_PUBLIC_CALL_STACK_LENGTH_PER_TX],
+
+    previous_validation_request_array_lengths: PublicValidationRequestArrayLengths, // Lengths of the validation requests before the enqueued call.
+    validation_requests: PublicValidationRequests, // Validation requests emitted throughout the enqueued call.
+    previous_accumulated_data_array_lengths: PublicAccumulatedDataArrayLengths, // Lengths of the accumulated data before the enqueued call.
+    accumulated_data: PublicAccumulatedData, // Data emitted throughout the enqueued call.
+    start_side_effect_counter: u32,
+    end_side_effect_counter: u32,
+    start_gas_left: Gas,
+    transaction_fee: Field,
+    reverted: bool,
```

Comment:
I wouldn't want to block all the goodness of this PR on something so minor. But if it's easy to have `end_gas_left` instead of `gas_used` in the VM public inputs, that would make our life slightly easier!

Reply:
Got it! Having `end_gas_left` also seems more straightforward. Will change it in another pr. Will also remove the `gas_used` in `PublicAccumulatedData` to avoid any confusion.